From 7dfea039dfdb6ba98cc84a0e19acb559f8450b4b Mon Sep 17 00:00:00 2001
From: Victoria Milhoan <vicki.milhoan@freescale.com>
Date: Thu, 20 Nov 2014 11:28:28 -0700
Subject: [PATCH 0903/5242] MLK-9951 Update CAAM driver era interface

commit  dd025dd5e0bc7640c59446a715f8d2f449654eaa from
https://source.codeaurora.org/external/imx/linux-imx.git

Add more CAAM era values to the CAAM driver's caam_get_era()
function. Read only 32 bits of data since the data required
to identify the IP_ID and MAJ_REV is located in the first 32
bits of the register. And, update the function for use with
ARM/Little Endian devices.

[<vicki.milhoan@freescale.com>: Edited to apply to 3.14]
Signed-off-by: Victoria Milhoan <vicki.milhoan@freescale.com>
(cherry picked from commit 6050d7faf2d0c063195aa9454c130548a9f8058f)
Signed-off-by: Dan Douglass <dan.douglass@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/ctrl.c |   15 ++++++++++++++-
 drivers/crypto/caam/regs.h |    4 ++++
 2 files changed, 18 insertions(+), 1 deletion(-)

diff --git a/drivers/crypto/caam/ctrl.c b/drivers/crypto/caam/ctrl.c
index ece1f28..fdef077 100644
--- a/drivers/crypto/caam/ctrl.c
+++ b/drivers/crypto/caam/ctrl.c
@@ -416,7 +416,20 @@ static int caam_get_era_from_hw(struct caam_ctrl __iomem *ctrl)
 		{0x0A11, 2, 5},
 		{0x0A12, 2, 5},
 		{0x0A13, 1, 5},
-		{0x0A1C, 1, 5}
+		{0x0A1C, 1, 5},
+		{0x0A12, 4, 6},
+		{0x0A13, 2, 6},
+		{0x0A16, 2, 6},
+		{0x0A17, 1, 6},
+		{0x0A18, 2, 6},
+		{0x0A1A, 1, 6},
+		{0x0A1C, 2, 6},
+		{0x0A14, 3, 7},
+		{0x0A10, 4, 8},
+		{0x0A11, 3, 8},
+		{0x0A11, 4, 8},
+		{0x0A12, 5, 8},
+		{0x0A16, 3, 8}
 	};
 	u32 ccbvid, id_ms;
 	u8 maj_rev, era;
diff --git a/drivers/crypto/caam/regs.h b/drivers/crypto/caam/regs.h
index 791ee5d..9203fa2 100644
--- a/drivers/crypto/caam/regs.h
+++ b/drivers/crypto/caam/regs.h
@@ -282,6 +282,10 @@ struct sec_vid {
 	u8 min_rev;
 };
 
+#define SEC_VID_IPID_SHIFT      16
+#define SEC_VID_MAJ_SHIFT       8
+#define SEC_VID_MAJ_MASK        0xFF00
+
 struct caam_perfmon {
 	/* Performance Monitor Registers			f00-f9f */
 	u64 req_dequeued;	/* PC_REQ_DEQ - Dequeued Requests	     */
-- 
1.7.9.5

